### PR TITLE
AJAX email uniqness checking in DB

### PR DESCRIPTION
**Pull request by Mikitsik**
Ticket: 
https://trello.com/c/uMe6BMQm

______
Add AJAX email uniq checking during new user registration

When new user reach modal registration window and try to write his
email he would see a blue hint alert for correct email writing. If he
would write a correct shape of email but this email would be already
exist in IWR-Project database he would see a red alert of this occur.
When his email would not be find in IWR-Project DB he would see
green announcement of this.
With this feature JS would check whether new user email write correctly due
to standart email pattern and rails would check his email uniqness
in DB.
A Register button is disabled while red and hint alerts occurs

### CODE REVIEW

**Comment by Mikitsik** [comment]
[[Feature] Add checking of email uniqness in DB on the fly while registration process](https://trello.com/c/uMe6BMQm/76-feature-add-checking-of-email-uniqness-in-db-on-the-fly-while-registration-process)

File: app/helpers/home_helper.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+module HomeHelper
```
Comment (gogochikago) [inline app/helpers/home_helper.rb]: Delete empty file

**Comment by Mikitsik** [comment]
Add AJAX email uniq checking during new user registration

When new user reach modal registration window and try to write his
email he will see a blue hint alert for correct email writing. If he
would write a correct shape of email but this email would be already
exist in IWR-Project database he would see a red alert of this occur.
When his email would not be find in IWR-Project DB he would see
green announcement of this.
With this feature JS would check whether new user email write correctly due
to standart email pattern and rails would check his email uniqness
in DB.
A Register button is disabled while red and hint alerts occurs

File: config/routes.rb [hunk 2]
```diff
@@ -8,6 +8,7 @@
   post '/login', to: 'sessions#create'
   delete '/logout', to: 'sessions#destroy'
 
+  get '/users/email', to: 'users#email_uniq?'
```
Comment (alexshgov) [inline config/routes.rb]: Let's be RESTful
Consider the following API endpoint: `GET /user/existence/:email`
```bash
Response if user exists
Status: 204 No Content
Response if user does not exist
Status: 404 Not Found
```

**Pull request merged**